commit 6b5a86f342e4b6b086f13b5b50d1eb1209a0045c
Author: Dave Mielke <Dave@Mielke.cc>
Date:   Mon Mar 8 18:31:49 2021 -0500

    Fix support for the APH Chameleon 20 with firmware version 1.1. (dm)

diff --git a/Drivers/Braille/HumanWare/braille.c b/Drivers/Braille/HumanWare/braille.c
index 8a2b2f233..1feb40fd0 100644
--- a/Drivers/Braille/HumanWare/braille.c
+++ b/Drivers/Braille/HumanWare/braille.c
@@ -946,34 +946,74 @@ connectResource (BrailleDisplay *brl, const char *identifier) {
       .data = &resourceData_HID_touch
     },
 
-    { /* APH Chameleon 20 (HID protocol) */
+    { /* APH Chameleon 20 (HID protocol, firmware 1.0) */
       .vendor=0X1C71, .product=0XC101, 
       .configuration=1, .interface=1, .alternative=0,
       .inputEndpoint=4, .outputEndpoint=5,
+      .verifyInterface = 1,
       .data = &resourceData_HID_C20,
       .resetDevice = 1
     },
 
-    { /* APH Mantis Q40 (HID protocol) */
+    { /* APH Chameleon 20 (HID protocol, firmware 1.1) */
+      .vendor=0X1C71, .product=0XC101, 
+      .configuration=1, .interface=0, .alternative=0,
+      .inputEndpoint=1, .outputEndpoint=2,
+      .verifyInterface = 1,
+      .data = &resourceData_HID_C20,
+      .resetDevice = 1
+    },
+
+    { /* APH Mantis Q40 (HID protocol, firmware 1.0) */
       .vendor=0X1C71, .product=0XC111, 
       .configuration=1, .interface=1, .alternative=0,
       .inputEndpoint=4, .outputEndpoint=5,
+      .verifyInterface = 1,
+      .data = &resourceData_HID_M40,
+      .resetDevice = 1
+    },
+
+    { /* APH Mantis Q40 (HID protocol, firmware 1.1) */
+      .vendor=0X1C71, .product=0XC111, 
+      .configuration=1, .interface=0, .alternative=0,
+      .inputEndpoint=1, .outputEndpoint=2,
+      .verifyInterface = 1,
       .data = &resourceData_HID_M40,
       .resetDevice = 1
     },
 
-    { /* NLS eReader (HID protocol) */
+    { /* NLS eReader (HID protocol, firmware 1.0) */
       .vendor=0X1C71, .product=0XCE01, 
       .configuration=1, .interface=1, .alternative=0,
       .inputEndpoint=4, .outputEndpoint=5,
+      .verifyInterface = 1,
       .data = &resourceData_HID_NLS,
       .resetDevice = 1
     },
 
-    { /* Humanware BrailleOne (HID protocol) */
+    { /* NLS eReader (HID protocol, firmware 1.1) */
+      .vendor=0X1C71, .product=0XCE01, 
+      .configuration=1, .interface=0, .alternative=0,
+      .inputEndpoint=1, .outputEndpoint=2,
+      .verifyInterface = 1,
+      .data = &resourceData_HID_NLS,
+      .resetDevice = 1
+    },
+
+    { /* Humanware BrailleOne (HID protocol, firmware 1.0) */
       .vendor=0X1C71, .product=0XC121, 
       .configuration=1, .interface=1, .alternative=0,
       .inputEndpoint=4, .outputEndpoint=5,
+      .verifyInterface = 1,
+      .data = &resourceData_HID_one,
+      .resetDevice = 1
+    },
+
+    { /* Humanware BrailleOne (HID protocol, firmware 1.1) */
+      .vendor=0X1C71, .product=0XC121, 
+      .configuration=1, .interface=0, .alternative=0,
+      .inputEndpoint=1, .outputEndpoint=2,
+      .verifyInterface = 1,
       .data = &resourceData_HID_one,
       .resetDevice = 1
     },
